pkg/virt-handler/vm.go

```diff
@@ -687,6 +690,31 @@ func (d *VirtualMachineController) migrationSourceUpdateVMIStatus(origVMI *v1.Vi
 	return nil
 }
 
+type NetworkStatus struct {
+	Name      string   `yaml:"name"`
+	Ips       []string `yaml:"ips"`
+	Interface string   `yaml:"interface"`
+}
+
+// Using the downward API, look for dedicated migration network net1 and, if found, set migration IP to its IP
+func findMigrationIP(vmi *v1.VirtualMachineInstance, migrationIp *string) {
```

Comment:
Can you move this whole `findMigrationIP` to the virt-handler startup phase and just set d.ipAddress upfront on the controller? It can not change during the handler lifetime and it shoud keep the core-logic more in the focus then.